src/Endpoint/AuthorizedEndpointBase.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenEuropa\CdtClient\Endpoint;
+
+use OpenEuropa\CdtClient\Contract\RestInterface;
+use OpenEuropa\CdtClient\Model\Response\Token;
+
+/**
+ * Class AuthorizedEndpointBase
+ *
+ * Serves as the base for endpoint classes with token support.
+ */
+abstract class AuthorizedEndpointBase extends EndpointBase
```

Comment:
After removing the container, I noticed that the `TokenAwareTrait` does not make the code any simpler. On the contrary, it requires a lot of extra lines. I changed it to a shorter solution.